pkg/monitor/monitor/util.go

```diff
@@ -772,17 +890,29 @@ func defaultTidbMonitor(monitor *v1alpha1.TidbMonitor) {
 		}
 		monitor.Spec.Clusters[id] = tcRef
 	}
+	if monitor.Spec.DMSpec != nil {
+		for id, dcRef := range monitor.Spec.DMSpec.Clusters {
+			if len(dcRef.Namespace) < 1 {
+				dcRef.Namespace = monitor.Namespace
+			}
+			monitor.Spec.DMSpec.Clusters[id] = dcRef
+		}
+	}
 	retainPVP := core.PersistentVolumeReclaimRetain
 	if monitor.Spec.PVReclaimPolicy == nil {
 		monitor.Spec.PVReclaimPolicy = &retainPVP
 	}
 }
 
-func getMonitorStatefulSet(sa *core.ServiceAccount, config *core.ConfigMap, secret *core.Secret, monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster) (*apps.StatefulSet, error) {
+func getMonitorStatefulSet(sa *core.ServiceAccount, config *core.ConfigMap, secret *core.Secret, monitor *v1alpha1.TidbMonitor, tc *v1alpha1.TidbCluster, dc *v1alpha1.DMCluster) (*apps.StatefulSet, error) {
 	statefulSet := getMonitorStatefulSetSkeleton(sa, monitor)
 	initContainer := getMonitorInitContainer(monitor, tc)
 	statefulSet.Spec.Template.Spec.InitContainers = append(statefulSet.Spec.Template.Spec.InitContainers, initContainer)
-	prometheusContainer := getMonitorPrometheusContainer(monitor, tc)
+	if dc != nil {
+		dmInitContainer := getMonitorDMInitContainer(monitor, dc, tc.TiDBImage())
```

Comment:
The version may not be `tc.TiDBImage()`.

Reply:
Fixed.